### PR TITLE
[smart-switch] Extend config generator for t1-smartswitch topology.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
The Smart Switch includes two types of devices (Switch and DPU) that require the sample configuration. Generate Smart Switch sample configurations for Switch or DPU based on the device type.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Extend the sample config generator for t1-smartswitch topology.

#### How to verify it
Run sonic-cfggen UT. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Comment by oleksandrivantsiv** [comment]
@prabhataravind kindly reminder. Please review the PR

File: src/sonic-config-engine/smartswitch_config.py [hunk 1]
```diff
@@ -14,25 +15,29 @@
 except KeyError:
     pass
 
-DPUS_TABLE  = 'DPUS'
+DPU_TABLE = 'DPU'
```
Comment (prabhataravind) [inline src/sonic-config-engine/smartswitch_config.py:18]: Why do we have these two?

File: src/sonic-config-engine/tests/sample_output/t1-smartswitch-dpu.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+    "DEVICE_METADATA": {
+        "localhost": {
+            "hwsku": "SS-DPU-1x400Gb",
+            "hostname": "sonic",
+            "switch_type": "dpu",
+            "type": "SonicDpu",
+            "subtype": "SmartSwitcch",
```
Comment (prabhataravind) [inline src/sonic-config-engine/tests/sample_output/t1-smartswitch-dpu.json:8]: typo here?

File: src/sonic-config-engine/config_samples.py [hunk 3]
```diff
@@ -75,42 +77,89 @@ def generate_t1_sample_config(data):
...
+
         data['DHCP_SERVER_IPV4_PORT'] = dhcp_server_ports
 
     return data
 
+def generate_t1_smartswitch_dpu_sample_config(data, ss_config):
+    data['DEVICE_METADATA']['localhost']['hostname'] = 'sonic'
+    data['DEVICE_METADATA']['localhost']['switch_type'] = 'dpu'
+    data['DEVICE_METADATA']['localhost']['type'] = 'SonicDpu'
+    data['DEVICE_METADATA']['localhost']['subtype'] = 'SmartSwitcch'
```
[the remark this answers is not in the document]
Comment (oleksandrivantsiv) [reply src/sonic-config-engine/config_samples.py:146]: Good catch. Thanks!

File: src/sonic-config-engine/smartswitch_config.py (hunk 1 above)
Comment (oleksandrivantsiv) [reply src/sonic-config-engine/smartswitch_config.py:18]: `DPUS` indicates that the image is running on the switch.
`DPU` indicates that the image is running on the DPU.
https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md#device-data-and-platform

Comment (oleksandrivantsiv) [reply src/sonic-config-engine/smartswitch_config.py:18]: Based on the type of device we can generate different configurations for switch and DPU for the same topology

**Review by prabhataravind — APPROVED** [review]

**Comment by oleksandrivantsiv** [comment]
@prabhataravind, @prsunny who can help to resolve ms-conflict?

**Comment by prabhataravind** [comment]
> @prabhataravind, @prsunny who can help to resolve ms-conflict?

@prsunny could you help here? This needs your approval as well.

**Review by prsunny — APPROVED** [review]

**Review by kperumalbfn — APPROVED** [review]

**Pull request merged**